botocore/discovery.py

```diff
@@ -87,14 +91,15 @@ def _gather_ids(self, shape, params, ids=None):
 
 
 class EndpointDiscoveryManager(object):
-    def __init__(self, client, cache=None, current_time=None):
+    def __init__(self, client, cache=None, current_time=None, strict=True):
         if cache is None:
             cache = {}
         self._cache = cache
         self._failed_attempts = {}
         if current_time is None:
             current_time = time.time
         self._time = current_time
+        self._strict = strict
```

Comment:
It might be worth renaming it to something else. Typically, I associate strict with causing validation errors. Maybe call it `always_discover`?